sale_invoice_plan/models/sale.py

```diff
@@ -0,0 +1,200 @@
+        percent = round(1.0 / num_installment * 100, prec)
+        percent_last = 100 - (percent * (num_installment-1))
+        # Advance
+        if advance:
+            vals = {'installment': 0, 'plan_date': installment_date,
+                    'type': 'advance', 'percent': 0.0}
+            invoice_plans.append((0, 0, vals))
+            installment_date = self._next_date(installment_date,
+                                               interval, interval_type)
+        # Normal
+        for i in range(num_installment):
+            this_installment = i+1
+            if num_installment == this_installment:
+                percent = percent_last
+            vals = {'installment': this_installment,
+                    'plan_date': installment_date,
+                    'type': 'installment',
+                    'percent': percent}
+            invoice_plans.append((0, 0, vals))
+            installment_date = self._next_date(installment_date,
+                                               interval, interval_type)
+        self.write({'invoice_plan_ids': invoice_plans})
+        return True
+
+    @api.multi
+    def remove_invoice_plan(self):
+        self.ensure_one()
+        self.invoice_plan_ids.unlink()
+        return True
+
+    @api.model
+    def _next_date(self, installment_date, interval, interval_type):
+        installment_date = fields.Date.from_string(installment_date)
+        if interval_type == 'month':
+            next_date = installment_date + relativedelta(months=+interval)
+        elif interval_type == 'year':
+            next_date = installment_date + relativedelta(years=+interval)
+        else:
+            next_date = installment_date + relativedelta(days=+interval)
+        next_date = fields.Date.to_string(next_date)
+        return next_date
+
+    @api.multi
+    def action_invoice_create(self, grouped=False, final=False):
+        inv_ids = super().action_invoice_create(grouped=grouped, final=final)
+        invoice_plan_id = self._context.get('invoice_plan_id')
+        if invoice_plan_id:
+            plan = self.env['sale.invoice.plan'].browse(invoice_plan_id)
+            invoices = self.env['account.invoice'].browse(inv_ids)
+            invoices.ensure_one()  # Expect 1 invoice for 1 invoice plan
+            plan._compute_new_invoice_quantity(invoices[0])
+            plan.invoice_ids += invoices
+
+        return inv_ids
+
+
+class SaleInvoicePlan(models.Model):
+    _name = 'sale.invoice.plan'
+    _description = 'Invoice Planning Detail'
+    _order = 'installment'
+
+    sale_id = fields.Many2one(
+        comodel_name='sale.order',
+        string='Sales Order',
+        index=True,
+        readonly=True,
+        ondelete='cascade',
+    )
+    installment = fields.Integer(
+        string='Installment',
+    )
+    plan_date = fields.Date(
+        string='Plan Date',
+        required=True,
+    )
+    type = fields.Selection(
```

Comment:
@kittiu IMO I think here change field name it looks like a function

Reply:
@nikul-serpentcs thanks for your comment.